geopandas/sindex.py

```diff
@@ -754,23 +765,36 @@ def query_bulk(self, geometry, predicate=None, sort=False):
         def nearest(
             self, geometry, return_all=True, max_distance=None, return_distance=False
         ):
-            if not compat.PYGEOS_GE_010:
-                raise NotImplementedError("sindex.nearest requires pygeos >= 0.10")
-
-            geometry = self._as_geometry_array(geometry)
+            if not (compat.USE_SHAPELY_20 or compat.PYGEOS_GE_010):
```

Comment:
```suggestion
            if not (compat.USE_SHAPELY_20 or (compat.USE_PYGEOS and compat.PYGEOS_GE_010)):
```

Reply:
In practice, this method only gets called if you have USE_PYGEOS (or USE_SHAPELY_20). Hmm, which on second thought means that I probably didn't even have to add `compat.USE_SHAPELY_20`